app/web/features/communities/CommunitiesPage/CommunitiesPage.tsx

```diff
@@ -0,0 +1,280 @@
+        {
+          data: query.data!.pages.flatMap((page) => page.communitiesList),
+          hasMore: query.hasNextPage,
+        },
+      ]);
+      setSelected([...selected.slice(0, level), community]);
+    } else {
+      if (community.communityId === selected[level].communityId) {
+        //a previously selected item is clicked, so unselect it
+        //treat level = 0 as a special case
+        if (level === 0) {
+          setSelected([]);
+        } else {
+          setSelected([...selected.slice(0, level)]);
+        }
+        setCachedQueryResults(cachedQueryResults.slice(0, level));
+      } else {
+        //a previously unselected item is clicked
+        setSelected([...selected.slice(0, Math.max(level, 0)), community]);
+        setCachedQueryResults(cachedQueryResults.slice(0, level + 1));
+      }
+    }
+  };
+
+  return (
+    <div>
+      <div className={classes.headerRow}>
+        <PageTitle>{t("communities:communities_title")}</PageTitle>
+      </div>
+      <Typography
+        variant="body1"
+        paragraph
+        className={classes.createCommunityText}
+      >
+        <Trans i18nKey="dashboard:your_communities_helper_text2">
+          {`Don't see your community? `}
+          <MuiLink
+            href={COMMUNITY_BUILDER_FORM_LINK}
+            target="_blank"
+            rel="noreferrer noopener"
+            underline="hover"
+          >
+            Get it started!
+          </MuiLink>
+        </Trans>
+      </Typography>
+      <Typography
+        variant="body1"
+        paragraph
+        className={classes.createCommunityText}
+      >
+        <Trans i18nKey="dashboard:community_guidelines_helper_text">
+          {`Learn more about how we build our communities! `}
+          <MuiLink
+            href={COMMUNITY_GUIDELINES_LINK}
+            target="_blank"
+            rel="noreferrer noopener"
+            underline="hover"
+          >
+            Read our community guidelines
+          </MuiLink>
+        </Trans>
+      </Typography>
+
+      <div className={classes.root}>
+        {cachedQueryResults.map((query, index) => (
+          <BrowserColumn
+            key={index}
+            parent={selected?.[index - 1] ?? globalCommunityQuery.data}
+            communities={query.data}
+            handleClick={(community) => handleClick(community, index)}
+            selected={selected[index]?.communityId}
+          />
+        ))}
+        {query.isLoading ? ( // div prevents overflow scrollbar from spinner
+          <div className={classes.loader}>
+            <CenteredSpinner />
+          </div>
+        ) : query.isSuccess && globalCommunityQuery.isSuccess ? (
+          <div ref={lastColumnRef}>
+            <BrowserColumn
+              parent={
+                selected?.[selected.length - 1] ?? globalCommunityQuery.data
+              }
+              communities={query.data.pages.flatMap(
+                (page) => page.communitiesList,
+              )}
```

Comment:
Would you mind putting the communities browser in its own component? You can call it `CommunitiesBrowser`, then import that component in here.

Separates the logic more nicely that way I think.